pymc3/sampling.py

```diff
@@ -530,23 +530,94 @@ def sample_ppc(trace, samples=None, model=None, vars=None, size=None,
 
     seed(random_seed)
 
-    if weights is not None:
-        if len(trace) != len(weights):
-            raise ValueError(
-                'The number of traces and weights should be the same')
+    indices = randint(0, len(trace), samples)
+    if progressbar:
```

Comment:
tqdm progress bars need to be closed if sampling is interrupted. Some try...finally would do it.